### PR TITLE
wgengine/magicsock: wait for endpoint updater goroutine when closing.

### PR DESCRIPTION
**Pull request by danderson**
I know you love sync.Conds :). Pondering another code structure that lets us avoid it, but this is the minimum necessary fix to get correct ordering of shutdown.

---
This change is [Reviewable](https://reviewable.io/reviews/tailscale/tailscale/332)

### CODE REVIEW

**Comment by danderson** [comment]
btw: checked this change with `go test -count=10000 -run TestDeviceStartStop tailscale.com/wgengine/magicsock` and `go test -race -count=10000 -run TestDeviceStartStop tailscale.com/wgengine/magicsock`. Without this change, both crash within a few hundred iterations due to the shutdown race. With this change, they can get through all 10k.

**Comment by bradfitz** [comment]
You're right that `sync.Cond` is generally terrible and makes code too hard to follow, but it's fine for now. We'll remove it later.
